<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Type-only imports allow to use a declaration defined elsewhere for type annotations and declarations.
Forcing type-only imports prevents issues where the file referenced in the import exports a type and the import is done for a value:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">// ./vehicles.ts
export type Car = /*...*/;

// ./main.ts
import { Car } from './vehicles';</code></pre>
</div>
</div>
<div class="paragraph">
<p>This would fail at runtime, because the export would be erased and the import remain, thus breaking the chain. This does not affect the latest Typescript transpiler, but may affect other transpilation tools such as the <a href="https://esbuild.github.io/">esbuild</a> bundler.</p>
</div>
<div class="paragraph">
<p>When using the imported declaration exclusively for typing annotations and declarations, use <code>import type</code> instead of <code>import</code>.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import { Car } from 'vehicles';
const x: Car = /*...*/;</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-javascript" data-lang="javascript">import type { Car } from 'vehicles';
const x: Car = /*...*/;</code></pre>
</div>
</div>
</div>
</div>